substrate/frame/contracts/src/storage.rs

```diff
@@ -164,13 +164,35 @@ impl<T: Config> ContractInfo<T> {
 		storage_meter: Option<&mut meter::NestedMeter<T>>,
 		take: bool,
 	) -> Result<WriteOutcome, DispatchError> {
-		let child_trie_info = &self.child_trie_info();
 		let hashed_key = key.hash();
+		self.write_raw(&hashed_key, new_value, storage_meter, take)
+	}
+
+	/// Update a storage entry into a contract's kv storage.
+	/// Function used in benchmarks, which can simulate prefix collision in keys.
+	#[cfg(feature = "runtime-benchmarks")]
+	pub fn bench_write_raw(
```

Comment:
Instead of this call we could add and use Key::Raw variant, but I am not sure if it is better option, WDYT?

Reply:
wouldn't call `child::put_raw` enough for the benchmark?

Reply:
I suggest keeping it this way. We can't use `child::put_raw` directly because that wouldn't include the read which is also part of every write due to storage deposits.